src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -1309,3 +1309,43 @@ def _get_statistics_for_weights_compression(
                 for node in matmul_nodes:
                     statistics[node.node_name] = copy.deepcopy(stats)
         return statistics
+
+    def get_config(self):
+        return {
+            "mode": self._mode,
+            "ratio": self._ratio,
+            "group_size": self._group_size,
+            "ignored_scope": self._ignored_scope,
+            "all_layers": self._all_layers,
+            "sensitivity_metric": self._sensitivity_metric,
+            "awq": self._awq,
+            "subset_size": self._subset_size,
+            "scale_estimation": self._scale_estimation,
+            "gptq": self._gptq,
+            "lora_correction": self._lora_correction,
+            "backup_mode": self._backup_mode,
+            "compression_format": self._compression_format,
+            "advanced_parameters": self._advanced_parameters,
```

Comment:
Please make a copy of mutable objects there and check it doesn't affect the original params in the test